### PR TITLE
rider build to use codeartifact to get nuget depencencies

### PR DESCRIPTION
**Pull request by vchikoti1998**
## Types of changes

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Motivation and Context
Added a condition in rider build script, If codeartifact available, should get nuget dependencies from it.
## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [x ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if change is customer facing in the IDE.
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .run/Run IDE - Rider [2022.2].run.xml [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
-    </configuration>
-</component>
+  <configuration default="false" name="Run IDE - Rider [2022.2]" type="GradleRunConfiguration" factoryName="Gradle" folderName="2022.2">
+    <log_file alias="idea.log" path="$PROJECT_DIR$/jetbrains-rider/build/idea-sandbox/system/log/idea.log" />
+    <ExternalSystemSettings>
+      <option name="env">
+        <map>
+          <entry key="CODEARTIFACT_NUGET_URL" value="https://my-domain-255356604037.d.codeartifact.us-west-2.amazonaws.com/nuget/my-repository/v3/index.json" />
+        </map>
+      </option>
```
Comment (rli) [inline .run/Run IDE - Rider [2022.2].run.xml:9]: should not be committed

File: buildspec/windowsTests.yml [hunk 2]
```diff
@@ -62,7 +63,7 @@ phases:
             copyFolder $root "build/test-results/test/" $script:TEST_REPORTS
         }
 
-        copyArtifacts "."
+        copyArtifacts "."
```
Comment (rli) [inline buildspec/windowsTests.yml:66]:
```suggestion
        copyArtifacts "."
```

File: buildspec/windowsTests.yml [hunk 3]
```diff
@@ -61,8 +62,7 @@ phases:
             copyFolder $root "build/idea-sandbox/system-test/log/" $script:TEST_ARTIFACTS
             copyFolder $root "build/test-results/test/" $script:TEST_REPORTS
         }
-
-        copyArtifacts "."
+        copyArtifacts "."
```
Comment (rli) [inline buildspec/windowsTests.yml:65]:
```suggestion

        copyArtifacts "."  
```

File: jetbrains-rider/build.gradle.kts [hunk 4]
```diff
@@ -10,8 +10,7 @@ import java.nio.file.Path
 
 buildscript {
     // Cannot be removed or else it will fail to compile
-    @Suppress("RemoveRedundantQualifierName")
-    val rdversion = software.aws.toolkits.gradle.intellij.IdeVersions.ideProfile(project).rider.rdGenVersion
+    @Suppress("RemoveRedundantQualifierName") val rdversion = software.aws.toolkits.gradle.intellij.IdeVersions.ideProfile(project).rider.rdGenVersion
```
Comment (rli) [inline jetbrains-rider/build.gradle.kts:13]:
```suggestion
    @Suppress("RemoveRedundantQualifierName")
    val rdversion = software.aws.toolkits.gradle.intellij.IdeVersions.ideProfile(project).rider.rdGenVersion
```

File: jetbrains-rider/build.gradle.kts [hunk 5]
```diff
@@ -64,11 +63,7 @@ val resharperPluginPath = File(projectDir, "ReSharper.AWS")
 val resharperBuildPath = File(project.buildDir, "dotnetBuild")
 
 val resharperParts = listOf(
-    "AWS.Daemon",
-    "AWS.Localization",
-    "AWS.Project",
-    "AWS.Psi",
-    "AWS.Settings"
+    "AWS.Daemon", "AWS.Localization", "AWS.Project", "AWS.Psi", "AWS.Settings"
```
Comment (rli) [inline jetbrains-rider/build.gradle.kts:66]:
```suggestion
    "AWS.Daemon",
    "AWS.Localization",
    "AWS.Project",
    "AWS.Psi",
    "AWS.Settings"
```

File: jetbrains-rider/build.gradle.kts [hunk 6]
```diff
@@ -196,10 +201,7 @@ val buildReSharperPlugin = tasks.register("buildReSharperPlugin") {
 
     doLast {
         val arguments = listOf(
-            "build",
-            "--verbosity",
-            "normal",
-            "${resharperPluginPath.canonicalPath}/ReSharper.AWS.sln"
+            "build", "--verbosity", "normal", "${resharperPluginPath.canonicalPath}/ReSharper.AWS.sln"
```
Comment (rli) [inline jetbrains-rider/build.gradle.kts:204]:
```suggestion
            "build",
            "--verbosity",
            "normal",
            "${resharperPluginPath.canonicalPath}/ReSharper.AWS.sln"
```

File: jetbrains-rider/build.gradle.kts [hunk 7]
```diff
@@ -175,11 +170,21 @@ val prepareNuGetConfig = tasks.register("prepareNuGetConfig") {
...
-    <add key="resharper-sdk" value="$nugetPath" />
-  </packageSources>
-</configuration>
+  <configuration>
+    <packageSources> 
+    ${
+        if (codeArtifactNugetUrl.isPresent) {
+            """
+       |   <clear /> 
+   |       <add key="codeartifact-nuget" value="${codeArtifactNugetUrl.get()}v3/index.json" />
```
Comment (rli) [inline jetbrains-rider/build.gradle.kts:179]:
```suggestion
       |   <clear /> 
       |   <add key="codeartifact-nuget" value="${codeArtifactNugetUrl.get()}v3/index.json" />
```

**Review by rli — COMMENTED** [review]
in general minor formatting changes are okay to include but it's unclear why any of these were done

**Comment by vchikoti1998** [comment]
> in general minor formatting changes are okay to include but it's unclear why any of these were done

It was done by running _./gradlew detekt_

File: jetbrains-rider/build.gradle.kts (hunk 7 above)
Comment (vchikoti1998) [reply jetbrains-rider/build.gradle.kts:179]: For the suggested change L 178 & L 179. 
Its not generating a proper xml format in Nuget.config file. 
Its generating in this way: 
<config>
   <pkg>
        <clear/>
    <add key= ""     />    
         <add ..................  />
   </pkg>
</config>

File: jetbrains-rider/build.gradle.kts [hunk 8]
```diff
@@ -174,14 +174,31 @@ val prepareNuGetConfig = tasks.register("prepareNuGetConfig") {
...
+            ""
+        }
+        }
+    </packageSources>
+  </configuration>
+"""
         val configText = """<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
+    <clear />
```
Comment (rli) [inline jetbrains-rider/build.gradle.kts:196]: ?
```suggestion
```

**Pull request merged**